File: zul/src/archive/web/js/zul/mesh/mold/paging.js
/* paging.js

	Purpose:
		
	Description:	
		
	History:
		Fri Jan 23 15:00:45     2009, Created by jumperchen

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

This program is distributed under LGPL Version 2.1 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/
function (out) {
	if (this.getMold() == "os") {
		out.push('<div', this.domAttrs_(), '>', this._innerTags(), '</div>');
		return;
	}
	
	var uuid = this.uuid,
		btn = this.$s('button');
	
	out.push('<div name="', uuid, '"', this.domAttrs_(), '><ul>',
			'<li><a name="', uuid, '-first" class="', btn, ' ', this.$s('first'),
				'" href="javascript:;"><i class="z-paging-icon z-icon-angle-double-left"></i></a></li>',
			'<li><a name="', uuid, '-prev" class="', btn, ' ', this.$s('previous'),
				'" href="javascript:;"><i class="z-paging-icon z-icon-angle-left"></i></a></li>',
			'<li><input name="',
				uuid, '-real" class="', this.$s('input'), '" type="text" value="',
				this.getActivePage() + 1, '" size="3"></input><span class="',
				this.$s('text'), '"> / ', this.getPageCount(), '</span></li>',
			'<li><a name="', uuid, '-next" class="', btn, ' ', this.$s('next'),
				'" href="javascript:;"><i class="z-paging-icon z-icon-angle-right"></i></a></li>',
			'<li><a name="', uuid, '-last" class="', btn, ' ', this.$s('last'),
				'" href="javascript:;"><i class="z-paging-icon z-icon-angle-double-right"></i></a></li></ul>');
	
	if (this.isDetailed())
		this._infoTags(out);
	out.push('</div>');
}

File: zul/src/org/zkoss/zul/api/Paging.java
/* Paging.java

	Purpose:
		
	Description:
		
	History:
		Tue Oct 22 14:45:31     2008, Created by Flyworld

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

{{IS_RIGHT
	This program is distributed under LGPL Version 3.0 in the hope that
	it will be useful, but WITHOUT ANY WARRANTY.
}}IS_RIGHT
 */
package org.zkoss.zul.api;

import org.zkoss.zul.ext.Paginal;

/**
 * Paging of long content.
 * 
 * <p>
 * Default {@link #getZclass}: z-paging. (since 3.5.0)
 * 
 * @author tomyeh
 * @since 3.5.2
 */
public interface Paging extends org.zkoss.zul.impl.api.XulElement, Paginal {

	/**
	 * Returns whether to automatically hide this component if there is only one
	 * page available.
	 * <p>
	 * Default: false.
	 */
	public boolean isAutohide();

	/**
	 * Sets whether to automatically hide this component if there is only one
	 * page available.
	 */
	public void setAutohide(boolean autohide);
}
